Jenkinsfile

```diff
@@ -16,8 +16,10 @@ pipeline {
     }
     stage('Run tests and archive test results') {
       steps {
-        sh './test.sh'
-        sh 'sudo chown -R jenkins:jenkins .'  // bad docker mount creates unreadable files TODO fix this
+        lock("api-java-${env.NODE_NAME}") {
+          sh './test.sh'
+          sh 'sudo chown -R jenkins:jenkins .'  // bad docker mount creates unreadable files TODO fix this
```

Comment:
We have the problem that when we run containers with mounts (via `docker` or `docker-compose`), we often end up with root-owned read-only files, instead of files owned by the user that called `docker` or `docker-compose`. So we are doing this, for now, in Jenkins to fix those permissions so that the builds can be auto-cleaned up.

I dunno if I agree with the conclusion here, but he lays out some possibilities.

https://denibertovic.com/posts/handling-permissions-with-docker-volumes/

Let's talk about this with @jvanderhoof and @apotterri in #about-docker Slack when you're free.